src/ucs/datastruct/mpool.c

```diff
@@ -295,7 +295,12 @@ ucs_status_t ucs_mpool_hugetlb_malloc(ucs_mpool_t *mp, size_t *size_p, void **ch
 
     /* First, try hugetlb */
     real_size = *size_p;
-    status = ucs_sysv_alloc(&real_size, real_size * 2, (void**)&ptr, SHM_HUGETLB,
+    status = ucs_sysv_alloc(&real_size, real_size * 2, (void**)&ptr,
+#ifdef SHM_HUGETLB
```

Comment:
what about define additional macro which has value SHM_HUGETLB in case if SHM_HUGETLB is defined and 0 in other cases. you may eliminate few #ifdef's in arguments